src/QsCompiler/Compiler/CompilationLoader.cs

```diff
@@ -324,8 +359,17 @@ public Status LoadedRewriteStep(string name, string source = null)
         /// Uses the specified logger to log all diagnostic events. 
         /// Throws an ArgumentNullException if either one of the given loaders is null or returns null.
         /// </summary>
-        public CompilationLoader(SourceLoader loadSources, ReferenceLoader loadReferences, Configuration? options = null, ILogger logger = null)
+        public CompilationLoader(SourceLoader loadSources, ReferenceLoader loadReferences, Configuration? options = null, ILogger logger = null, CompilationTaskEventHandler onCompilationTaskEvent = null)
         {
+
+            // adding the event handlers.
+            if (onCompilationTaskEvent != null)
+            {
+                CompilationTaskEvent += onCompilationTaskEvent;
```

Comment:
I think it is a good idea. The code will look cleaner since the event handler wouldn't have to be passed to the constructor. I will make the changes to make this member static.